src/Components/QuickGrid/Microsoft.AspNetCore.Components.QuickGrid/src/QuickGrid.razor

```diff
@@ -63,12 +63,15 @@
         }
     }
 
+
     private void RenderRow(RenderTreeBuilder __builder, int rowIndex, TGridItem item)
     {
-        <tr @key="@(ItemKey(item))" aria-rowindex="@rowIndex">
+        <tr @key="@(ItemKey(item))" aria-rowindex="@rowIndex" @onclick="@(async () => await HandleRowClick(item))">
```

Comment:
I know it's painful but unfortunately, for performance reasons, this is a case where we'd need to fall back on raw RenderTreeBuilder logic instead of using Razor syntax.

Having an `@onclick` unconditionally like this means that (1) we're always doing several extra allocations per row even if you're not using this feature (because you're building a delegate regardless), and (2) we're always assigning and removing event IDs for every row even if you're not using this feature.

To avoid these problems, we'd need to change the `RenderRow` method so that the outer `<tr>` is constructed through a series of `__builder.*` calls, with the "onclick" added inside an `if (OnRowClicked.HasDelegage) { ... }`. The contents within that (i.e., the `@foreach (var col in _columns)` and so on) can continue to be emitted using Razor syntax.

Does that make sense?